src/device/rdk/output/image.rs

```diff
@@ -1,207 +1,161 @@
-            let response = ErrorResponse {
-                status: 400,
-                error: "Error parsing request: ".to_string() + err.to_string().as_str(),
-            };
-            let Response_json = json!(response);
-            return Err(serde_json::to_string(&Response_json).unwrap());
+    let rt = Runtime::new().unwrap();
+    rt.block_on(async {
+        let mut ResponseOperator = CaptureScreenshotResponse::default();
+        // *** Fill in the fields of the struct CaptureScreenshotResponse here ***
+        
+        //######### Enable the Http server #########
+        let my_local_ip = local_ip().unwrap();
+        let my_server: String = "http://".to_string() + &my_local_ip.to_string() + &":7878/upload".to_string();
+        let addr = SocketAddr::from(([0, 0, 0, 0], 7878));
+        let (tx, mut rx) = mpsc::channel(10);
+
+        let make_svc = make_service_fn(move |_conn| {
+            let tx = tx.clone();
+            async move { Ok::<_, Infallible>(service_fn(move |req| handle_req(req, tx.clone()))) }
+        });
+
+        let (shutdown_tx, shutdown_rx) = oneshot::channel();
+
+        let server = Server::bind(&addr).serve(make_svc);
+        let graceful = server.with_graceful_shutdown(async {
+            shutdown_rx.await.ok();
+        });
+
+        tokio::spawn(graceful);
+
+
+        //######### Activate org.rdk.ScreenCapture #########
+        service_activate("org.rdk.ScreenCapture".to_string()).unwrap();
+
+        //#########org.rdk.ScreenCapture.uploadScreenCapture#########
+        #[derive(Serialize)]
+        struct UploadScreenCaptureRequest {
+            jsonrpc: String,
+            id: i32,
+            method: String,
+            params: UploadScreenCaptureRequestParams,
         }
-        Ok(_) => (),
-    }
 
-    let Dab_Request: OutputImageRequest = IncomingMessage.unwrap();
+        #[derive(Serialize)]
+        struct UploadScreenCaptureRequestParams {
+            url: String,
+            callGUID: String,
+        }
 
-    if Dab_Request.outputLocation.is_empty() {
-        let response = ErrorResponse {
-            status: 400,
-            error: "request missing 'outputLocation' parameter".to_string(),
+        let req_params = UploadScreenCaptureRequestParams {
+            url: my_server,
+            callGUID: "12345".to_string(),
         };
-        let Response_json = json!(response);
-        return Err(serde_json::to_string(&Response_json).unwrap());
-    }
-
-    service_activate("org.rdk.ScreenCapture".to_string()).unwrap();
 
-    //#########org.rdk.ScreenCapture.uploadScreenCapture#########
-    #[derive(Serialize)]
-    struct UploadScreenCaptureRequest {
-        jsonrpc: String,
-        id: i32,
-        method: String,
-        params: UploadScreenCaptureRequestParams,
-    }
+        let request = UploadScreenCaptureRequest {
+            jsonrpc: "2.0".into(),
+            id: 3,
+            method: "org.rdk.ScreenCapture.uploadScreenCapture".into(),
```

Comment:
The result of this method doesn't mean anything right?
Is there a way to do event listener on uploadComplete?

Also, after looking at the API. It's so much easier if we just upload it to the DAB host directly. Sigh....

Reply:
- Is there a way to do event listener on uploadComplete?
@bgy36ww , yes, there is, using websockets. It will return the response + the uploadComplete. However, uploadComplete is triggered after the data is sent to HTTP server. In the code I am waiting the data arrive in HTTP server (and timing out when it not) , so the uploadComplete is not important.
- Also, after looking at the API. It's so much easier if we just upload it to the DAB host directly. Sigh....
Yes... =( At least for DAB. I don´t know if most of other hw partners works the same way.